### PR TITLE
#18795 Adding live:true condition automatically when Time Machine is …

### PR DESCRIPTION
**Pull request by freddyucv**
When a lucene query is pass to the method:

https://github.com/dotCMS/core/blob/b91e4ecc94e8fe652653cefc5f3eec266b0df382/dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/ContentTool.java#L155

is called with a lucene query that not has the live:true condition and Time Machine is running, now we are adding the live:true condition automatically

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java [hunk 1]
```diff
@@ -414,4 +423,176 @@ private ContentTool getContentTool(final long languageId){
...
+        final ViewContext viewContext = mock(ViewContext.class);
+        when(viewContext.getRequest()).thenReturn(request);
+
+        contentTool.init(viewContext);
+        return contentTool;
+    }
+
+    /**
+     * Method to test: {@link ContentTool#pullPerPage(String, int, int, String)}
+     * When: there is a content with a expire  date set to tomorrow and the time machine date is the date after tomorrow
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java:507]: code expresses a different thing that this javadoc

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package com.dotcms.util;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.rules.model.Condition;
+import com.dotmarketing.portlets.rules.model.ConditionGroup;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:8]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.ConditionGroup'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package com.dotcms.util;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
+import com.dotmarketing.exception.DotDataException;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:5]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
 import com.dotcms.contenttype.model.type.SimpleContentType;
 import com.dotcms.datagen.ContentletDataGen;
 import com.dotcms.datagen.TestDataUtils;
+import com.dotcms.rendering.velocity.viewtools.content.util.ContentUtils;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java:20]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.rendering.velocity.viewtools.content.util.ContentUtils'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import com.dotmarketing.exception.DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:6]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import com.dotmarketing.portlets.rules.model.Condition;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:7]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.rules.model.Condition'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.dotcms.util;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+
+import javax.servlet.http.HttpSession;
+import java.util.Optional;
+
+public class TimeMachineUtil {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java:8]: Codacy found an issue: [A class which only has private constructors should be final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
+package com.dotcms.util;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:4]: Codacy found an issue: [Avoid unused imports such as 'com.dotcms.enterprise.rules.RulesAPIImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TimeMachineUtilTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:26]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
+package com.dotcms.util;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.enterprise.rules.RulesAPIImpl;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.rules.model.Condition;
+import com.dotmarketing.portlets.rules.model.ConditionGroup;
+import com.liferay.portal.model.User;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.portal.model.User'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import java.util.Date;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:19]: Codacy found an issue: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/viewtools/content/ContentToolTest.java:506]: done https://github.com/dotCMS/core/pull/18863/commits/a73fc1f7ed898352425c4be3a6211ffafdea42a6#diff-b77a02567ac2b8e298742516ca666f0dR510

File: dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java (hunk 7 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java:8]: Doc

File: dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+    private TimeMachineUtil(){}
+
+    public static Optional<String> getTimeMachineDate() {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java:12]: doc

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java:12]: shouldn't be getSession(false)

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TimeMachineUtilTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import java.util.Date;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java (hunk 12 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/util/TimeMachineUtil.java:16]: done https://github.com/dotCMS/core/pull/18863/commits/de4603169a1241fa8698a2dc200bd71efdb03d2b

**Comment by erickgonzalez** [comment]
There is a test failiing that seems related to Time Machine https://raw.githack.com/dotCMS/test-results/issue-18795-Time-machine-Widgets-are-not-respecting-the-publish-expire-dates/projects/core/current/integration/postgres/reports/html/classes/com.dotmarketing.cms.urlmap.URLMapAPIImplTest.html#shouldReturnURLInfoWhenContentExistsInFutureButTimeMachineIsSet

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

**Comment by wezell** [comment]
@DeanGonzalez You are the product owner on this - we can give you a docker image to test against.

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:20]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

File: dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/util/TimeMachineUtilTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5835963)

**Pull request merged**